torch/distributed/nn/functional.py

```diff
@@ -443,7 +443,7 @@ class _AllReduce(Function):
     def forward(ctx, op, group, tensor):
         ctx.group = group
         ctx.op = op
-        tensor = tensor.clone()
+        tensor = tensor.clone().contiguous()
```

Comment:
```suggestion
        tensor = tensor.clone(memory_format= torch.contiguous_format)
```
Only need to do one copy

Reply:
Actually, this would be a great torchfix check, I see this a few other places in the codebase